components/data-provider/src/structs/plurals.rs

```diff
@@ -0,0 +1,52 @@
+// Plural types
+use serde::{Deserialize, Serialize};
+use std::borrow::Cow;
+
+#[cfg(feature = "invariant")]
+use crate::prelude::*;
+
+/// Gets a locale-invariant default struct given a data key in this module's category.
+#[cfg(feature = "invariant")]
+pub(crate) fn get_invariant(data_key: &DataKey) -> Option<DataResponse<'static>> {
+    use crate::invariant::make_inv_response;
+    if data_key.category != DataCategory::Plurals {
+        return None;
+    }
+    match data_key.sub_category.as_str() {
```

Comment:
This nested match block with a good level of symmetery looks like a good candidate to condense by [matching on a tuple](https://stackoverflow.com/a/39160179).